### PR TITLE
fix: reject invalid identity zone

### PR DESCRIPTION
**Pull request by klaus-sap**
JdbcScimUserProvisioning.create() and update() so far accepted invalid identity zone ids
Needed to adapt two test classes to fix this

### CODE REVIEW

File: server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java [hunk 1]
```diff
@@ -551,17 +550,15 @@ public void updateLastLogonTime(String id, String zoneId) {
...
 
-    @Nullable
-    private UserConfig getUserConfig(String zoneId) {
+    private UserConfig getUserConfig(String zoneId) throws InvalidScimResourceException {
         try {
             IdentityZone currentZone = identityZoneManager.getCurrentIdentityZone();
             return (currentZone.getId().equals(zoneId)) ?
-                currentZone.getConfig().getUserConfig() :
-                jdbcIdentityZoneProvisioning.retrieve(zoneId).getConfig().getUserConfig();
+                    currentZone.getConfig().getUserConfig() :
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: please do not add TABs

File: server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java [hunk 2]
```diff
@@ -551,17 +550,15 @@ public void updateLastLogonTime(String id, String zoneId) {
...
+                    jdbcIdentityZoneProvisioning.retrieve(zoneId).getConfig().getUserConfig();
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: TAB

**Review by strehle — APPROVED** [review]

**Comment by klaus-sap** [comment]
https://github.com/cloudfoundry/uaa/pull/2712 has been merged

**Review by adrianhoelzl-sap — APPROVED** [review]

**Review by Tallicia — APPROVED** [review]
Looks good, I like the more specific exception usage and the additional test as well as including zone with the zoneID.

**Comment by Tallicia** [comment]
Is this ready to merge? Or is there any prerequisite this depends on before merging?

**Pull request merged**